### PR TITLE
docs: Improve AWS SQS examples & troubleshooting

### PR DESCRIPTION
**Pull request by preflightsiren**
_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #

* Adds troubleshooting information for AWS environments.
* Adds an example for AWS scalers using podIdentity.

### CODE REVIEW

File: content/troubleshooting/aws-keda-operator-iam.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
++++
+title = "Why does KEDA operator error with NoCredentialProviders"
++++
+
+If you are running KEDA on aws using IRSA or KIAM for pod identity and seeing the following error messages:
```
Comment (tomkerkhove) [inline content/troubleshooting/aws-keda-operator-iam.md:5]: Nit
```suggestion
If you are running KEDA on AWS using IRSA or KIAM for pod identity and seeing the following error messages:
```

File: content/troubleshooting/aws-keda-operator-iam.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  Type     Reason                      Age                From           Message
+  ----     ------                      ----               ----           -------
+  Normal   KEDAScalersStarted          31s                keda-operator  Started scalers watch
+  Normal   KEDAScaleTargetDeactivated  31s                keda-operator  Deactivated apps/v1.Deployment default/my-event-based-deployment from 1 to 0
+  Normal   ScaledObjectReady           13s (x2 over 31s)  keda-operator  ScaledObject is ready for scaling
+  Warning  KEDAScalerFailed            1s (x2 over 31s)   keda-operator  NoCredentialProviders: no valid providers in chain. Deprecated.
+           For verbose messaging see aws.Config.CredentialsChainVerboseErrors
+```
+
+and the operator logs:
```
Comment (tomkerkhove) [inline content/troubleshooting/aws-keda-operator-iam.md:18]: Nit
```suggestion
And the operator logs:
```

File: content/troubleshooting/aws-keda-operator-iam.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+2021-11-02T23:50:29.688Z    ERROR    controller    Reconciler error    {"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledObject", "controller": "scaledobject", "name": "my-event-based-deployment-scaledobject", "namespace": "default"
+, "error": "error getting scaler for trigger #0: error parsing SQS queue metadata: awsAccessKeyID not found"}
+```
+
+This means keda-operator is not receiving valid credentials - even before attempting to assume the IAM role associated with the `scaleTargetRef`.
```
Comment (tomkerkhove) [inline content/troubleshooting/aws-keda-operator-iam.md:25]: Nit
```suggestion
This means that the KEDA operator is not receiving valid credentials, even before attempting to assume the IAM role associated with the `scaleTargetRef`.
```

File: content/troubleshooting/aws-keda-operator-iam.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Some things to check:
+
+* Ensure the keda-operator deployment has the `iam.amazonaws.com/role` annotation under `deployment.spec.template.metadata` not `deployment.metadata` - if using KIAM
```
Comment (tomkerkhove) [inline content/troubleshooting/aws-keda-operator-iam.md:29]:
```suggestion
* Ensure the `keda-operator` deployment has the `iam.amazonaws.com/role` annotation under `deployment.spec.template.metadata` not `deployment.metadata` - if using KIAM
```

File: content/troubleshooting/aws-keda-operator-iam.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+* Ensure the keda-operator serviceAccount is annotated `eks.amazonaws.com/role-arn` - if using IRSA
```
Comment (tomkerkhove) [inline content/troubleshooting/aws-keda-operator-iam.md:30]: Nit
```suggestion
* Ensure the `keda-operator` service account is annotated `eks.amazonaws.com/role-arn` - if using IRSA
```

File: content/troubleshooting/aws-keda-operator-iam.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+* Check kiam-server logs, successful provisioning of credentials looks like:
```
Comment (tomkerkhove) [inline content/troubleshooting/aws-keda-operator-iam.md:31]: I presume this is an AWS-specific deployment or can you elaborate a bit on this please?
```suggestion
* Check `kiam-server` logs, successful provisioning of credentials looks like:
```

File: content/troubleshooting/aws-keda-operator-iam.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+`kube-system kiam-server-6bb67587bd-2f47p kiam-server {"level":"info","msg":"found role","pod.iam.role":"arn:aws:iam::1234567890:role/my-service-role","pod.ip":"100.64.7.52","time":"2021-11-05T03:13:34Z"}`.
+`grep`ping for the keda-operator pod ip can help ensure keda-operator is successfully getting credentials.
```
Comment (tomkerkhove) [inline content/troubleshooting/aws-keda-operator-iam.md:33]: This feels off - Is this a new item or just extension that you should ping for the pod IP?

Maybe this should be something like this then?

```suggestion
  * ping for the keda-operator pod ip can help ensure keda-operator is successfully getting credentials.
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
Thanks a ton! Just added a few nits/questions to improve readability

**Review by zroubalik — APPROVED** [review]
Exceleent, thanks a ton!

File: content/troubleshooting/aws-keda-operator-iam.md (hunk 7 above)
Comment (preflightsiren) [reply content/troubleshooting/aws-keda-operator-iam.md:33]: I was trying to express that when you use the term "grepping" it's a conjoining of `grep` the tool and the suffix ping.

I'll formalise the language a bit.

**Comment by preflightsiren** [comment]
Addressed the PR feedback, hopefully the list renders better now :P

**Review by tomkerkhove — APPROVED** [review]
Thank you!

**Pull request merged**